src/runtime_src/xdp/profile/plugin/ocl/xocl_profile_cb.cpp

```diff
@@ -178,6 +228,8 @@ cb_action_ndrange(xocl::event* event,cl_int status,const std::string& cu_name, c
     std::string uniqueName = "KERNEL|" + uniqueDeviceName + "|" + xname + "|" + CuInfo + "|";
     std::string traceString = uniqueName + std::to_string(workGroupSize);
     OCLProfiler::Instance()->getPlugin()->setTraceStringForComputeUnit(kname, traceString);
+    // Log Buffers associated with this kernel
+    log_buffer_guidance(event, kernel);
```

Comment:
Please make sure this is called only once per kernel.

Reply:
line 158 addresses that already

Reply:
@hackwa Please move the checking code to the beginning of the log_buffer_guidance function and add a comment so it's 1. an early exit; and 2. much clearer that the function only operates once per kernel.

Reply:
done